### PR TITLE
NC-636

### PR DESCRIPTION
**Pull request by Joeper214**
Fixed update triggers when there is no trigger ids and sky zabbix update trigger call exception

### CODE REVIEW

File: app/models/zabbix.rb [hunk 1]
```diff
@@ -159,15 +162,20 @@ def update_trigger_expression(infra, trigger_exprs)
...
-        updating_expr = trigger_exprs[item["key_"]].sub("HOSTNAME", r.physical_id)
-        {
-          triggerid: item["triggers"].first["triggerid"],
-          expression: updating_expr
-        }
+        if item["triggers"].any?
+          updating_expr = trigger_exprs[item["key_"]].sub("HOSTNAME", r.physical_id)
+          {
+            triggerid: item["triggers"].first["triggerid"],
+            expression: updating_expr
```
Comment (skyarch-networks) [inline app/models/zabbix.rb]: __[rubocop]__
* Style/TrailingCommaInLiteral: Put a comma after the last item of a multiline hash. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-array-commas)

File: app/models/zabbix.rb [hunk 2]
```diff
@@ -159,15 +162,20 @@ def update_trigger_expression(infra, trigger_exprs)
...
+          }
+        end
       end
     end.flatten
 
-    @sky_zabbix.trigger.update(triggers)
+    begin
+      yield @sky_zabbix.trigger.update(triggers)
+    rescue => ex
```
Comment (skyarch-networks) [inline app/models/zabbix.rb]: __[rubocop]__
* Lint/HandleExceptions: Do not suppress exceptions. [:ref](https://github.com/bbatsov/ruby-style-guide#dont-hide-exceptions)
* Lint/UselessAssignment: Useless assignment to variable - `ex`. [:ref](https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)

**Comment by skyarch-networks** [comment]
Click here if you want to create rubocop's auto-correct Pull Request 
[](https://sideci.com/github_repositories/31754199/auto_correct/rubocop/pull_request/278/branch/NC-636)

See the documentation if you want to get more information about auto-correct.
[Document](https://sideci.com/ja/docs/autofix-rubocop)

**Comment by skyarch-networks** [comment]
Created Auto-correct Pull Request from [SideCI](https://sideci.com/)

[https://github.com/skyarch-networks/skyhopper/pull/279](https://github.com/skyarch-networks/skyhopper/pull/279)

File: app/models/zabbix.rb [hunk 3]
```diff
@@ -159,15 +162,21 @@ def update_trigger_expression(infra, trigger_exprs)
...
+          }
+        end
       end
     end.flatten
 
-    @sky_zabbix.trigger.update(triggers)
+    # Verify what caused the errors. @candy
+    begin
+      yield @sky_zabbix.trigger.update(triggers)
+    rescue
```
Comment (skyarch-networks) [inline app/models/zabbix.rb]: __[rubocop]__
* Lint/HandleExceptions: Do not suppress exceptions. [:ref](https://github.com/bbatsov/ruby-style-guide#dont-hide-exceptions)

**Comment by Joeper214** [comment]
Please Review and Merge PR @root-beer

**Comment by root-beer** [comment]
I'll review it now.

**Comment by root-beer** [comment]
LGTM. Thanks!

**Pull request merged**